lib/rules/require-direct-export.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * @fileoverview require the component to be directly exported
+ * @author Hiroki Osame <hiroki.osame@gmail.com>
+ */
+'use strict'
+
+const utils = require('../utils')
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'require the component to be directly exported',
+      category: 'essential',
```

Comment:
We add every new rule as `uncategorized` to have a nice period for testing it and introducing breaking changes if necessary, without having to wait for another major release. So I think we should change the category to `undefined` for the time being :) We're still in v5 beta though, so it might land in v5 after all - we'll see.

In theory it's perfectly fine to have the component assigned to a variable and then exported. You can use a special comment for determining that this is a Vue component and it will be linted properly:
```js
// @vue/component
const ComponentA = {
  ...
}

export default ComponentA;
```

The way I see it - essentials are here to prevent errors. And this is technically not an error-situation so I think it would better fit in `strongly-recommended` :)

Reply:
Ahh I didn't know about the comment. Good to know, thank you!